neurons/miner.py

```diff
@@ -192,452 +206,520 @@ def __init__(self):
-            target=self.block_listener,
-            args=(self.loop,),
-            daemon=True,
-        )
-        self.listener.start()  #
-        # Load Peers
-        if not self.config.peers:
-            self.peers = self.comms.peers
-            tplr.logger.info(f"Filtered gather peers with buckets: {self.peers}")
-        else:
-            self.peers = self.config.peers
-        if self.uid not in self.peers:
-            self.peers.append(self.uid)
-
-        self.comms.commitments = await self.comms.get_commitments()
-        self.comms.update_peers_with_buckets()
-        tplr.logger.info("Loaded commitments")
-
-        # Fetch start_window from highest stake validator
-        self.start_window = await self.comms.get_start_window()
-        tplr.logger.info(f"Using start_window: {self.start_window}")
-
-        self.global_step = self.current_window - self.start_window
-        tplr.logger.info(f"starting at Global Step : {self.global_step}")
-
-        # Proceed to load checkpoint
-        (
-            success,
-            loaded_momentum,
-            loaded_global_step,
-            loaded_optimizer,
-            loaded_scheduler,
-        ) = await self.comms.load_checkpoint(
-            model=self.model,
-            optimizer=self.optimizer,
-            scheduler=self.scheduler,
-            transformer=self.transformer,
-            compressor=self.compressor,
-            current_window=self.current_window,
-            device=self.config.device,
-            peers=self.peers,
-            uid=self.uid,
-            totalks=self.totalks,
-        )
-        if success:
-            self.momentum = loaded_momentum
-            self.global_step = loaded_global_step
-            self.optimizer = loaded_optimizer
-            self.scheduler = loaded_scheduler
-            tplr.logger.info(
-                f"Loaded checkpoint with global_step={self.global_step}, "
-                f"optimizer_step={self.optimizer.state_dict()['state'].get(0, {}).get('step', 0)}, "
-                f"scheduler_step={self.scheduler.last_epoch}"
-            )
-        else:
-            tplr.logger.info("Starting from scratch")
-            self.momentum = {
-                n: torch.zeros_like(p) for n, p in self.model.named_parameters()
-            }
-            self.model.to(self.config.device)
-
-        self.comms.start_commitment_fetcher()
-        self.comms.start_background_tasks()
-
-        while True:
-            # 1. Initialize window and update peers
-            window_start = tplr.T()
-            # Start the gather in the background:
-            gather_start = tplr.T()
-            step_window = self.current_window
-            self.global_step = (
-                self.current_window - self.start_window
-            )  # Update global_step
-            tplr.logger.info(
-                f"\n{'-' * 40} Window: {step_window} (Global Step: {self.global_step}) {'-' * 40}"
+        try:
```

Comment:
Since you are doing `miner.cleanup` in `if __name__ == "__main__":` do we really need this try/finally here as well?